### ISSUE TITLE
setgenerate multiple times result in deadlock

### ISSUE THREAD

**Issue by jvandenbroek** [Gulden/gulden-official#178]
Platform: Linux amd64 (normal tar release)
Interface: QT

When executing Gulden-cli setgenerate true multiple times (to set different no threads) results in locking up Gulden and can only be killed with SIGKILL.

The debug log just displays terminated as last line before it stops responding:

```
2019-10-23 10:02:44 GuldenGenerate terminated
2019-10-23 10:02:44 GuldenGenerate started
2019-10-23 10:02:44 keypool reserve 1660
...
2019-10-23 10:02:57 GuldenGenerate terminated
```

Also tried to setgenerate false first before re-enabling, same result.

**Comment by jvandenbroek** [comment]
I've figured out what's going on. You already had a 'fixme' in for the minerThread interrupt not getting there in time, so it will hang until a new block is created. For the UI you just spawn that in a new thread, so I tried the same method the RPC call, which indeed no longer locks up the main thread. But would be nice if you'd find a way to reliable interrupt the threads w/o waiting for next block? But I suppose that's already on your list :)
